### PR TITLE
fix: Fix the referer url

### PR DESCRIPTION
**Pull request by harrisonliu5**
Signed-off-by: harrisonliu5 <harrisonliu_5@163.com>

### What type of PR is this?

/kind bug

### Which issue(s) this PR fixes:

Fixes #

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note
fix the referer URL that it is have '&'
```

### Additional documentation, usage docs, etc.:

```docs

```

/assign @wansir

### CODE REVIEW

File: server/controllers/view.js [hunk 1]
```diff
@@ -45,7 +45,8 @@ const renderView = async ctx => {
 }
 
 const renderLogin = async ctx => {
-  ctx.cookies.set('referer', ctx.query.referer)
+  const referer = ctx.querystring.split('referer=')[1]
+  ctx.cookies.set('referer', referer || ctx.query.referer)
```
Comment (wansir) [inline server/controllers/view.js:49]:
```suggestion
if (isValidReferer(referer)) {
  ctx.cookies.set('referer', referer)
}

```

Comment (harrisonliu5) [reply server/controllers/view.js]: update

**Comment by wansir** [comment]
Related issue: https://github.com/kubesphere/kubesphere/issues/3378

**Pull request merged**